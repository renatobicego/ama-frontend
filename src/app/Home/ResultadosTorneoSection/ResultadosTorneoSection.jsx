import Image from "next/image"
import Link from "next/link"

const ResultadosTorneoSection = () => {
    return(
        <section className="size-section flex gap-8 md:gap-14 lg:gap-20 my-20 lg:my-32 items-center h-[200px] lg:h-auto">
            <div className="relative w-1/2 h-full">
                <Image 
                    src='/imgs/cavem.jpg' 
                    width={1000} 
                    height={1000} 
                    alt="Imagen decorativa de torneos" 
                    className="w-full h-full object-cover max-h-[360px]"
                />
                <Image 
                    width={500} 
                    height={500} 
                    className="absolute w-3/4 md:w-[45%] h-1/2 -bottom-4 -right-4 p-2 bg-white object-cover" 
                    src='/imgs/murialdo.jpg' 
                    alt="Imagen decorativa de torneos" 
                />
            </div>
            <div className="flex flex-col gap-4 items-start justify-evenly w-1/2 h-full">
                <h3 className="font-title md:!leading-relaxed font-semibold text-title w-full md:w-60 lg:w-80 text-xl lg:text-3xl">
                    Resultados de Torneos y Calendario
                </h3>

                <Link href={"/torneos"}>
                    <button className="btn-primary">Consultar Torneos</button>
                </Link>
            </div>
        </section>
    )
}

export default ResultadosTorneoSection