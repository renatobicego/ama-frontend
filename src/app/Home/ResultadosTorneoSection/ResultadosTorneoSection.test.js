import '@testing-library/jest-dom/extend-expect'
import { cleanup, render, screen } from '@testing-library/react'
import ResultadosTorneoSection from './ResultadosTorneoSection'

describe('<ResultadosTorneoSection />', () => {
    
    let resultadosTorneos

    beforeEach(() => {
        resultadosTorneos = render(<ResultadosTorneoSection />)
    })

    afterEach(() => {
        cleanup()
    })
    
    test('renders content', () => {
        expect(resultadosTorneos.container).toMatchSnapshot()

        // Render title
        const title = screen.getByText('Resultados de Torneos y Calendario')
        expect(title).toBeInTheDocument()

        //Render button to visit section
        const button = screen.getByText('Consultar Torneos')
        expect(button).toBeInTheDocument()

        //Render imgs
        const imgs = screen.getAllByAltText('Imagen decorativa de torneos')
        expect(imgs.length).toBe(2)
    })
})