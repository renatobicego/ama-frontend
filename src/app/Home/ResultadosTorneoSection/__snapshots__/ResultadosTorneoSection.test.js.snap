// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultadosTorneoSection /> renders content 1`] = `
<div>
  <section
    class="size-section flex gap-8 md:gap-14 lg:gap-20 my-20 lg:my-32 items-center h-[200px] lg:h-auto"
  >
    <div
      class="relative w-1/2 h-full"
    >
      <img
        alt="Imagen decorativa de torneos"
        class="w-full h-full object-cover"
        data-nimg="1"
        decoding="async"
        height="1000"
        loading="lazy"
        src="/_next/image?url=%2Fimgs%2Ftorneos1.jpg&w=2048&q=75"
        srcset="/_next/image?url=%2Fimgs%2Ftorneos1.jpg&w=1080&q=75 1x, /_next/image?url=%2Fimgs%2Ftorneos1.jpg&w=2048&q=75 2x"
        style="color: transparent;"
        width="1000"
      />
      <img
        alt="Imagen decorativa de torneos"
        class="absolute w-3/4 md:w-[45%] h-1/2 -bottom-4 -right-4 p-2 bg-white object-cover"
        data-nimg="1"
        decoding="async"
        height="500"
        loading="lazy"
        src="/_next/image?url=%2Fimgs%2Ftorneos2.jpg&w=1080&q=75"
        srcset="/_next/image?url=%2Fimgs%2Ftorneos2.jpg&w=640&q=75 1x, /_next/image?url=%2Fimgs%2Ftorneos2.jpg&w=1080&q=75 2x"
        style="color: transparent;"
        width="500"
      />
    </div>
    <div
      class="flex flex-col gap-4 items-start justify-evenly w-1/2 h-full"
    >
      <h3
        class="font-title md:!leading-relaxed font-semibold text-title w-full md:w-60 lg:w-80 text-xl lg:text-3xl"
      >
        Resultados de Torneos y Calendario
      </h3>
      <a
        href="/torneos"
      >
        <button
          class="btn-primary"
        >
          Consultar Torneos
        </button>
      </a>
    </div>
  </section>
</div>
`;
